File: Source/Matlab_Test/testopen.m
% simple test open/close

function [status] = testopen(host)

  if(nargin < 1) host = 'localhost';
  end
  
  clientname = 'testopen';

  % source
  src = javaObject('com.rbnb.sapi.Source');
  src.OpenRBNBConnection(host,clientname);
  
  checkname = src.GetClientName;
  if(strcmp(checkname, clientname) == 0) 
      error('testopen: unexpected client name');
  end

  src.GetServerName;    % just for fun
  
  src.CloseRBNBConnection(0, 0);
  
  % sink
  snk = javaObject('com.rbnb.sapi.Sink');
  snk.OpenRBNBConnection(host,clientname);

  % ask for Registration map a couple ways
  cmap = javaObject('com.rbnb.sapi.ChannelMap');
  cmap.Add('_Log/...');
  snk.RequestRegistration(cmap); % request just Log chans
  snk.Fetch(10000);

  snk.RequestRegistration;       % request all chans
  snk.Fetch(10000);
  
  snk.CloseRBNBConnection(0, 0);
  
  fprintf('PASS: open/close test\n');
return

File: Source/JUnitTesting/src/com/rbnb/mtest/testopen.java
package com.rbnb.mtest;

import static org.junit.Assert.assertEquals;
import org.junit.Test;
import com.rbnb.sapi.*;

public class testopen extends testrbnb {
	String clientName = getClass().getSimpleName();

	@Test 
	public void thisTest() {
		assertEquals(clientName, true, runtest(clientName));
	}

	public boolean test(String host) throws SAPIException {
	// source
		Source src = new Source();
		src.OpenRBNBConnection(host,clientName);
		if(!src.GetClientName().equals(clientName)) {
			throw new SAPIException("unexpected client name");
		}
		src.GetServerName();    			// just for fun
		src.CloseRBNBConnection();
	// sink
		Sink snk = new Sink();
		snk.OpenRBNBConnection(host, clientName);
		ChannelMap cmap = new ChannelMap();
		cmap.Add("_Log/...");
		snk.RequestRegistration(cmap);		// request just log channels
		snk.Fetch(10000);
		snk.RequestRegistration();			// request all chans
		snk.Fetch(10000);
		snk.CloseRBNBConnection();
		return true;
	}
}
